### ISSUE TITLE
nspawn populates PATH without sbin directories

### ISSUE THREAD

**Issue by fgrsnau** [systemd/systemd#8698]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> systemd 238
> +PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN +PCRE2 default-hierarchy=hybrid

### Used distribution

> Arch Linux

### In case of bug report: Expected behaviour you didn't see

I’m currently trying to use `mkosi` on Arch Linux. I am hitting a problem where the script fails because the `sbin` directories are not part of the search path inside the container (when not running a login shell inside the container). 

### In case of bug report: Unexpected behaviour you saw

This boils down to `systemd-nspawn` setting `PATH=/usr/local/bin:/bin` by default which looks incorrect.

Here is a minimal example for reproducing the problem (on Arch Linux!):

```
# debootstrap stable root
[...]
# systemd-nspawn --directory=root /usr/bin/env
PATH=/usr/local/bin:/usr/bin:/bin
```

I would expect `PATH` to be something like `/usr/local/sbin:/usr/local/bin/:/usr/sbin:/usr/bin:/sbin/bin`. I think this is related to <https://github.com/systemd/systemd/blob/671f0f8de0a623cb3e4f66b562eae332135c8ca2/src/basic/path-util.h#L31> as `HAVE_SPLIT_BIN` is probably defined on my Arch Linux installation (at least `/usr/sbin` links to `/usr/bin` etc.).

Note that `systemd-nspawn` already tries to set a sane default PATH here: <https://github.com/systemd/systemd/blob/87ca11cb9201055348b4202210b06988c5a6ed9b/src/nspawn/nspawn.c#L2284>

Unfortunately, `DEFAULT_PATH_SPLIT_USR` depends on a specific `HAVE_SPLIT_BIN` definition and introduces a compatibility issue between distributions that have a split bin and ones that have not.

**Comment by poettering** [comment]
Hmm, yeah, i figure we should set the most generic PATH for our container payload indeed...

**Comment by poettering** [comment]
Fix waiting in #8756
